manager/manager.go

```diff
@@ -108,81 +105,79 @@ func (m *Manager) Close() (err error) {
 	return m.transactions.Close()
 }
 
-// New creates a transaction for the given config
-func (m *Manager) New(config transaction.Config) (tx transaction.Transaction, err error) {
-
-	d, ok := m.drivers.Load(config.Driver)
+// Run the given transaction
+func (m *Manager) Run(c transaction.Config) (r transaction.Result) {
+	var err error
 
-	if !ok {
-		return nil, fmt.Errorf("manager: transaction driver %s not registered", config.Driver)
-	}
+	uuid := ksuid.New().String()
+	start := time.Now()
 
-	if config.Inputs != nil {
-		if config, err = tmpl.Parse(config); err != nil {
-			return nil, fmt.Errorf("manager: error parsing transaction template: %w", err)
+	if c.Inputs != nil {
+		if c, err = tmpl.Parse(c); err != nil {
+			return wrapErrorResult(
+				uuid, c, start, fmt.Errorf("manager: error parsing transaction template: %w", err))
 		}
 	}
 
-	if config.Timeout == "" {
-		config.Timeout = DefaultTransactionTimeout
+	buf, err := json.Marshal(&c)
+	if err != nil {
+		return wrapErrorResult(
+			uuid, c, start, fmt.Errorf("manager: error marshaling config: %w", err))
 	}
 
-	return d.(driver.Driver).New(config)
-}
+	req, err := http.NewRequest(http.MethodPost, m.executorURL+uuid, bytes.NewReader(buf))
+	if err != nil {
+		return wrapErrorResult(
+			uuid, c, start, fmt.Errorf("manager: error creating executor request: %w", err))
+	}
 
-func (m *Manager) schedule(tx transaction.Transaction) (err error) {
+	res, err := m.client.Do(req)
+	if err != nil {
+		return wrapErrorResult(
+			uuid, c, start, fmt.Errorf("manager: error sending executor request: %w", err))
+	}
+	defer res.Body.Close()
 
-	var listener callback.Listener
-	config := tx.Config()
+	buf, err = ioutil.ReadAll(res.Body)
+	if err != nil {
+		return wrapErrorResult(
+			uuid, c, start, fmt.Errorf("manager: error reading executor response: %w", err))
+	}
 
-	if config.CallBack != nil {
-		listener, err = callback.GetListener(config.CallBack.Type)
-		if err != nil {
-			return err
-		}
+	err = json.Unmarshal(buf, &r)
+	if err != nil {
+		return wrapErrorResult(
+			uuid, c, start, fmt.Errorf("manager: error reading executor response: %w", err))
 	}
 
+	return r
+}
+
+func (m *Manager) schedule(config transaction.Config) (err error) {
+
 	return m.scheduler.AddTaskFunc(config.Name, config.Schedule,
 		func() {
 			var result transaction.Result
-			u := ksuid.New()
-			uuid := u.String()
-
-			for x := 0; x <= config.RetryCount; x++ {
-
-				ctx := context.WithValue(context.Background(), "transaction_uuid", uuid)
-				if config.CallBack != nil {
-					ctx = context.WithValue(ctx, config.CallBack.Type, listener)
-				}
-
-				result = tx.Run(ctx)
-				result.RetryCount = x
 
-				if !result.Failed {
+			var x int
```

Comment:
Why do we need `x` variable here? Does this even work with `x` uninitialized?

Reply:
Sure. we need the count outside the loop scope.